nemo/collections/asr/parts/submodules/rnnt_beam_decoding.py

```diff
@@ -198,7 +239,27 @@ def __init__(
         self.tsd_max_symmetric_expansion_per_step = tsd_max_sym_exp_per_step
         self.alsd_max_target_length = alsd_max_target_len
         self.nsc_max_timesteps_expansion = nsc_max_timesteps_expansion
-        self.nsc_prefix_alpha = nsc_prefix_alpha
+        self.nsc_prefix_alpha = int(nsc_prefix_alpha)
+        self.maes_prefix_alpha = int(maes_prefix_alpha)
+        self.maes_num_steps = int(maes_num_steps)
+        self.maes_expansion_gamma = float(maes_expansion_gamma)
+        self.maes_expansion_beta = int(maes_expansion_beta)
+
+        if self.maes_prefix_alpha < 0:
+            raise ValueError("`maes_prefix_alpha` must be a positive integer.")
+
+        if self.maes_num_steps < 1:
```

Comment:
I would suggest changing the condition to only allow `maes_num_steps >=2` if you don't apply any expansion regularization during training. Otherwise, you'll need to allow incomplete batch hypotheses to be carried onto the next step in order to tackle performance degradation for some edge cases when `maes_num_steps = 1`. Additionally, you may want to rely on a counting method to compute an `auto_nstep` during the last validation phase to improve the condition, i.e. only allows incomplete hypothesis when `maes_num_steps = 1` and `auto_nstep = 1`

Reply:
Sounds good, will set the limit ! Thanks for the great implementation !